app/models/project.rb

```diff
@@ -37,7 +37,16 @@ def repo_name
   end
 
   def docker_repo
-    "#{Rails.application.config.samson.docker.registry}/samson/#{repo_name}"
+    @docker_repo ||= begin
+      registry = Rails.application.config.samson.docker.registry
+      if Rails.env.production?
+        registry
+      else
+        host, namespace = registry.split '/'
+        namespace ||= 'samson'
```

Comment:
this is the least amount of work that allows to not overwrite each other's dev pushes. Do you think there is value in that? The proper solution would be to run a local registry.